include/arch/arm/arch.h

```diff
@@ -153,6 +153,23 @@ extern "C" {
 		sym[size+MPU_GUARD_ALIGN_AND_SIZE]
 #endif
 
+/**
+ * @brief Calculate size of stacks to be allocated in a stack array
+ *
+ * This macro calculates the size to be allocated for the stacks
+ * inside a stack array. It accepts the indicated "size" as a parameter
+ * and if required, pads some extra bytes (e.g. for MPU scenarios). Refer
+ * K_THREAD_STACK_ARRAY_DEFINE definition to see how this is used.
+ *
+ * @param size Size of the stack memory region
+ */
+#if defined(CONFIG_USERSPACE) && \
+	defined(CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT)
+#define _ARCH_THREAD_STACK_LEN(size) (POW2_CEIL(size))
+#else
+#define _ARCH_THREAD_STACK_LEN(size) (size+MPU_GUARD_ALIGN_AND_SIZE)
```

Comment:
Macro hygine: put `size` in parentheses.  Always always always parenthesize macro arguments expanded in C expressions.

Consider someone doing `K_THREAD_STACK_LEN(1 << my_shift_value)`.  That expands to:

`1 << my_shift_value + MPU_GUARD_ALIGN_AND_SIZE`

Which per the precedence rules evaluates as:

`1 << (my_shift_value + MPU_GUARD_ALIGN_AND_SIZE)`

Which needless to say is not what you want.  Always, always, always parenthesize macro arguments expanded into C expressions.

Reply:
embarrassingly stupid mistake. Will correct.